vllm/worker/model_runner.py

```diff
@@ -617,6 +685,16 @@ def prepare_input_tensors(
                 lora_mapping = None
 
             # Broadcast the metadata.
+            # If batch contains both prefill and decode, it sends 2 broadcasts.
+            # If it only contains 1 type, it triggers a single broadcast.
+            if (prefill_attn_metadata is not None
+                    and decode_attn_metadata is not None):
+                batch_type = BatchType.MIXED
+            elif prefill_attn_metadata is not None:
+                batch_type = BatchType.PREFILL
+            else:
+                batch_type = BatchType.DECODE
```

Comment:
Why is this needed? Can we detect this in the attention kernel instead?

Reply:
This is to send both metadata. Other alternative is to coelesce them in a single attention metadata